app/controllers/floating_ip_controller.rb

```diff
@@ -14,4 +14,260 @@ def self.display_methods
+      checked_floating_ip_id = checked_floating_ips[0] if checked_floating_ips.length == 1
+    end
+    checked_floating_ip_id
+  end
+
+  def networks_by_ems
+    assert_privileges("floating_ip_new")
+    networks = []
+    available_networks = CloudNetwork.where(:ems_id => params[:id], :external_facing => true).find_each
+    available_networks.each do |network|
+      networks << { 'name' => network.name, 'id' => network.id }
+    end
+    render :json => {
+      :available_networks => networks
+    }
+  end
+
+  def new
+    assert_privileges("floating_ip_new")
+    @floating_ip = FloatingIp.new
+    @in_a_form = true
+    @ems_choices = {}
+    ExtManagementSystem.where(:type => "ManageIQ::Providers::Openstack::NetworkManager").find_each do |ems|
+      @ems_choices[ems.name] = ems.id
+    end
+    @cloud_tenant_choices = {}
+    CloudTenant.all.each { |tenant| @cloud_tenant_choices[tenant.name] = tenant.id }
+    drop_breadcrumb(
+      :name => _("Add New Floating IP"),
+      :url  => "/floating_ip/new"
+    )
+  end
+
+  def update
+    assert_privileges("floating_ip_edit")
+    @floating_ip = find_by_id_filtered(FloatingIp, params[:id])
+
+    case params[:button]
+    when "cancel"
+      cancel_action(_("Edit of Floating IP \"%{name}\" was cancelled by the user") % { :name  => @floating_ip.name })
+
+    when "save"
+      options = form_params
+      options.delete(:ems_id)
+      task_id = @floating_ip.update_floating_ip_queue(session[:userid], options)
+
+      add_flash(_("Floating IP update failed: Task start failed: ID [%{id}]") %
+                {:id => task_id.inspect}, :error) unless task_id.kind_of?(Fixnum)
+
+      if @flash_array
+        javascript_flash(:spinner_off => true)
+      else
+        initiate_wait_for_task(:task_id => task_id, :action => "update_finished")
+      end
+    end
+  end
+
+  def update_finished
+    task_id = session[:async][:params][:task_id]
+    floating_ip_id = session[:async][:params][:id]
+    floating_ip_name = session[:async][:params][:name]
+    task = MiqTask.find(task_id)
+    if MiqTask.status_ok?(task.status)
+      add_flash(_("Floating IP \"%{name}\" updated") % { :name  => floating_ip_name })
+    else
+      add_flash(_("Unable to update Floating IP \"%{name}\": %{details}") % {
+        :name    => floating_ip_name,
+        :details => task.message }, :error)
+    end
+
+    @breadcrumbs.pop if @breadcrumbs
+    session[:edit] = nil
+    session[:flash_msgs] = @flash_array.dup if @flash_array
+
+    javascript_redirect :action => "show", :id => floating_ip_id
+  end
+
+  private
+
+  def form_params
+    options = {}
+    options[:ems_id] = params[:ems_id] if params[:ems_id] && params[:ems_id] != 'new'
+    options[:floating_ip_address] = params[:floating_ip_address] if params[:floating_ip_address]
+    options[:cloud_network_id] = params[:cloud_network_id] if params[:cloud_network_id]
+    options[:cloud_tenant_id] = params[:cloud_tenant_id] if params[:cloud_tenant_id]
```

Comment:
This needs to be something like:

options[:cloud_tenant] = find_by_id_filtered(CloudTenant, params[:cloud_tenant_id]) if params[:cloud_tenant_id]

Because the model expects options to include :cloud_tenant, and not :cloud_tenant_id (I'm currently fixing this in other places)